### DISCUSSION TITLE
BC Support: please provide the Pimcore's PHP Templating plugin to the comunnity.

### DISCUSSION THREAD

**Discussion by kanevbg** [pimcore/pimcore#10450] [General]
### Improvement description

It is not cool to leave the community without that bridge when you have it as it makes the upgrade **of the CORE** harder as it is big BC. This in result will lower the upgraded installations significantly I believe, and this is bad for the core and the pimcore community as whole. Please consider to make it available for the community.

**Comment by brusch** [comment]
Sorry, first of all this is not the right place -> moved to Discussions. 
Second, as every product we have to evolve as well and so we did. Actually, if you take it exactly, it was Symfony's decision to drop PHPEngine support 😉

Anyway, we are aware that this is a huge change, therefore we did this as part of a major release and for our enterprise-grade customers we're offering the PhpEngine as part of our commercial enterprise subscription (which is actually really not expensive) also for Pimcore 10 to gain more time for migrating the templates to Twig.

**Comment by kanevbg** [comment]
So I was asking to make the PhpEngine support free and open source in order to give more time to the people who chosen pimcore because it was free to update their system in the meantime and stay up to date & secure.

**Comment by kanevbg** [comment]
Yes, PHPEngine support is major release but you see that after releasing Pimcore 10 you provide no support support for version 6 so pimcore not given enough time (only ~1.5 months of support) in which to stay up to date and secure which is the significant thing. Am I right? It would be acceptable if 6.x was still supported actively but it is not supported at all.

**Reply by fashxp** [reply]
> It was deprecated since https://github.com/pimcore/pimcore/pull/7218, october 2020 ... So much more time than 1.5 months

**Reply by kanevbg** [reply]
@fashxp okay this is right. However I never seen and acknowledged this deprecation even I had bunch of Pimcore usability, projects and practice. This is how happened - I had missed that deprecation. Still do not understand why you should make my upgrade process harder than needed with forcing me to do the templating change, when you have solution for this, likely in big parts from the Version 6.x? I say again, if pimcore 6.x was at least maintained for security than it will be fine but now I am forced to upgrade.

**Comment by dpfaffenbauer** [comment]
Pimcore X is actually a big upgrade anyway and you should make the switch to Twig. You should not have started with PHP Engine in the first place like I told people since the release of Pimcore 5 and Symfony ;). If you cannot afford that, it is not too difficult to port the PHP engine for your project back yourself.

**Comment by kanevbg** [comment]
@dpfaffenbauer when it is not too difficult why should Pimcore not support its open source users in staying up to date and instead is willing to charge them for that big relieve in the upgrade.

**Comment by kanevbg** [comment]
If pimcore is not willing to release it I will do such add-on for the community just let me know the final opinion.

**Reply by brusch** [reply]
Sorry, this is definitely our final word on this, but let me quickly explain why: 
Pimcore is open-source, this is our legacy, our passion and we do by far most of our work for our free community edition and the whole ecosystem and community. But we're also a company & real people that need to earn some money, ... I guess you know what I mean. We cannot stop you from implementing such a bundle yourself and release it as open-source, but with that you're not really supporting the Pimcore ecosystem, you're actively working against it and put our future on risk - that doesn't help anybody long-term. 

To be honest, if you look at the budget for most of the projects being build with Pimcore, our pricing for enterprise support (LTS, ...) is really more than fair ... 😉
